app/controllers/problem_reports_controller.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class ProblemReportsController < ApplicationController
+  def index
+    @problem_reports = ProblemReport.order(id: :desc)
+    @filter = filter_message
+    if @filter == 'Unfixed and Critical'
+      @problem_reports = @problem_reports.where(is_critical: true,
+                                                is_fixed: false)
+    elsif @filter == 'Unfixed and Fixable'
+      @problem_reports = @problem_reports.where(is_fixable: true,
+                                                is_fixed: false)
+    end
+  end
+
+  def new; end
+
+  def create
+    if create_new_report.save
+      redirect_to problem_reports_path, notice: 'Success'
+    else
+      redirect_to new_problem_report_path, notice: 'Fail'
+    end
+  end
+
+  def update
+    @report = ProblemReport.find(params[:id])
+    update_remarks if params[:remarks]
+    update_bool_attr
+    redirect_to problem_reports_path
+  end
+
+  private
+
+  def update_remarks
+    @report.update(last_update_user_id: current_user.id,
+                   remarks: params[:remarks])
+  end
+
+  def update_bool_attr
+    %w[is_fixable is_fixed is_blocked is_critical].each do |a|
```

Comment:
I suggest:

1. Define a new method `boolean_params` and inside put `params.permit(:is_fixable, :is_fixed, ...)`
2. You can then just `@report.update(boolean_params.map { |k, v| [k, !v] })`

Reply:
params doesn't have map method

Reply:
There are methods like transform_key and transform_value, but I still need both key and value, because the update I've done is flipping if it is trigerred